charts/k8ssandra/templates/_helpers.tpl

```diff
@@ -117,24 +117,41 @@ Return the ingress host that should be used for Stargate's Cassandra/CQL interfa
 
 {{/*
 Create the jvm options based on heap properties specified.
+Expecting that c* heap.size and heap.newGenSize are NOT in IEC format.
 */}}
 {{- define "k8ssandra.configureJvmHeap" -}}
 {{- $datacenter := (index .Values.cassandra.datacenters 0) -}}
 {{- if $datacenter.heap }}
   {{- if $datacenter.heap.size }}
-      initial_heap_size: {{ $datacenter.heap.size }}
-      max_heap_size: {{ $datacenter.heap.size }}
+      {{- if (regexMatch ".+(i)" $datacenter.heap.size) }}
```

Comment:
I did a quick test and I think you might want to use the `print` function to make sure the values are cast to strings. I used the following chart properties:

```yaml
cassandra:
  version: 4.0.0
  heap:
    size: 1024
  datacenters:
  - name: dc1
```

The `helm install` did fail, but not as expected. Here is the error I got:

`
Error: template: k8ssandra/templates/cassandra/cassdc.yaml:148:4: executing "k8ssandra/templates/cassandra/cassdc.yaml" at <include "k8ssandra.configureJvmHeap" .>: error calling include: template: k8ssandra/templates/_helpers.tpl:142:38: executing "k8ssandra.configureJvmHeap" at <.Values.cassandra.heap.size>: wrong type for value; expected string; got float64
`

Reply:
Right, good find!  It looks through the tests we have at current are using a [string]string for value inputs, so that is how it fell through the cracks.  I'll add that print function to the regex functions for the cassdc values passed in.

Reply:
Updated w/ fix for this one and tested locally with latest.